packages/shorebird_cli/lib/src/commands/channels/delete_channels_command.dart

```diff
@@ -0,0 +1,125 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/auth_logger_mixin.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template delete_channels_command}
+/// `shorebird channels delete`
+/// Delete an existing channel for a Shorebird app.
+/// {@endtemplate}
+class DeleteChannelsCommand extends ShorebirdCommand
+    with AuthLoggerMixin, ShorebirdConfigMixin {
+  /// {@macro delete_channels_command}
+  DeleteChannelsCommand({
+    required super.logger,
+    super.buildCodePushClient,
+    super.auth,
+  }) {
+    argParser
+      ..addOption(
+        _appIdOption,
+        help: 'The app id that contains the channel to be deleted.',
+      )
+      ..addOption(
+        _channelNameOption,
+        help: 'The name of the channel to delete.',
+      );
+  }
+
+  static const String _appIdOption = 'app-id';
+  static const String _channelNameOption = 'name';
+
+  @override
+  String get description => 'Delete an existing channel for a Shorebird app.';
+
+  @override
+  String get name => 'delete';
+
+  @override
+  Future<int>? run() async {
```

Comment:
Can this be non-optional?

Reply:
Yeah was just keeping it consistent but we can adjust across the board